### PR TITLE
Integrate new p2p repo version

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request 
-  some tests that computed the amount of memory consumed sometimes failed. This was due to the fact that unnecessary libp2p components were running due to the poor code decoupling

## Proposed Changes
- integrated new p2p version
- removed code un-necessary dependency
- reverted some tests values

## Testing procedure
- standard system test
- half network test

### CODE REVIEW

File: node/external/timemachine/fee/memoryFootprint/memory_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 // keep this test in a separate package as to not be influenced by other the tests from the same package
 func TestFeeComputer_MemoryFootprint(t *testing.T) {
 	numEpochs := 10000
-	maxFootprintNumBytes := 30_000_000
+	maxFootprintNumBytes := 20_000_000
```
Comment (andreibancioiu) [inline node/external/timemachine/fee/memoryFootprint/memory_test.go:18]: :+1:

**Review by andreibancioiu — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]
don't forget proper release

**Comment by iulianpascalau** [comment]
> don't forget proper release

done

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by andreibancioiu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**